components/scream/src/share/atm_process/atmosphere_process.cpp

```diff
@@ -412,6 +388,49 @@ get_internal_field(const std::string& field_name) const {
   return get_internal_field_impl(field_name);
 }
 
+void AtmosphereProcess::
+add_property_check (const prop_check_ptr& pc)
```

Comment:
how about `add_invariant_property_check` or `add_inv_property_check`, to indicate that this checks an invariant instead of a pre/post condition (while we're using design-by-contract jargon)?

Reply:
I think that would be the most appropriate word. Do you think most people would get the meaning (i.e., invariant=pre+post)?

Reply:
I think if they looked up "invariant" and "precondition", they'd find the definition. An invariant is a predicate that holds for the entire lifetime of an object.

Reply:
Then I think the fully spelled out `add_invariant_property_check` might be better, since `inv` might not easily hint to "invariant"...